libs/libc/stdlib/lib_exit.c

```diff
@@ -22,13 +22,40 @@
  * Included Files
  ****************************************************************************/
 
+#include <nuttx/config.h>
+
+#include <nuttx/atexit.h>
+#include <nuttx/compiler.h>
+
 #include <stdlib.h>
 #include <unistd.h>
 
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+/* Put the weak definition of __dso_handle here */
+
+FAR void *__dso_handle weak_data;
```

Comment:
`libs/libxx/libxx.hxx` has `extern "C" FAR void *__dso_handle weak_data;` and that seems to be the difference that I'm talking about. So it is `extern` `weak` and not just `weak` like it is here.

Reply:
Correct

Reply:
I'm wondering if the declaration is even necessary